.github/workflows/main.yaml

```diff
@@ -0,0 +1,92 @@
+name: Main
+on: [push, pull_request]
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      
+    - name: Checkout
+      uses: actions/checkout@master
+
+    - name: Build
+      run: docker-compose -f ./cicd/docker/docker-compose-sdk.yaml build
+
+    - name: Build Release
+      if: |
+        github.event_name == 'push' && (
+          startsWith(github.event.ref, 'refs/tags/')
+          || endsWith(github.event.ref, '/master')
+        )
+      run: docker-compose -f ./cicd/docker/docker-compose-release.yaml build
+
+    - name: Compute Docker Tag
+      if: |
+        github.event_name == 'push' && (
+          startsWith(github.event.ref, 'refs/tags/')
+          || endsWith(github.event.ref, '/master')
+        )
+      id: compute_docker_tag
+      run: |
+        tag=${GITHUB_REF#refs/tags/}
+        branch=${GITHUB_REF#refs/heads/}
+        if [ "$tag" != "$GITHUB_REF" ]; then
+          tag=$(echo "$tag" | sed -e 's/[^a-zA-Z0-9\-\.]/-/g')
+          echo ::set-output name=DOCKER_TAG::${tag}
+        elif [ "$branch" != "$GITHUB_REF" ]; then
+          branch=$(echo "$branch" | sed -e 's/[^a-zA-Z0-9\-\.]/-/g')
+          echo ::set-output name=DOCKER_TAG::${branch}
+        else
+          echo "unable to determine docker tag" >&2
+          exit 1
+        fi
+
+    - name: Docker Login
+      if: |
+        github.event_name == 'push' && (
+          startsWith(github.event.ref, 'refs/tags/')
+          || endsWith(github.event.ref, '/master')
+        )
+      run: |
+        echo "${{ secrets.REGISTRY_PASSWORD }}" \
+          | docker login -u "${{ secrets.REGISTRY_USER }}" --password-stdin
+
+    - name: Docker Tag
+      if: |
+        github.event_name == 'push' && (
+          startsWith(github.event.ref, 'refs/tags/')
+          || endsWith(github.event.ref, '/master')
+        )
+      run: |
+        docker tag \
+          hobbyfarm/admin-ui:${GIT_COMMIT_SHORT_HASH:-dev} \
+          hobbyfarm/admin-ui:${{ steps.compute_docker_tag.outputs.DOCKER_TAG }}
+
+    - name: Docker Push
+      if: |
+        github.event_name == 'push' && (
+          startsWith(github.event.ref, 'refs/tags/')
+          || endsWith(github.event.ref, '/master')
```

Comment:
We are moving to only pushing images on `release/x.x.x` branches. Can we refactor this section to not perform a `docker push` unless on a release branch? 

I would also prefer not to auto-push all tags, but auto-pushing release/x.x.x tags is fine.

Reply:
With release branching wouldn't you still tag? I.e. cut a `release/2.1` branch and then tag 2.1.0 on it. Fix a bug then tag 2.1.1 on it.

Otherwise the full semver tags will always be overwritten when fixes are pushed to a release branch?

In that example it would make sense to push a continuous `2.1-rc-latest` tag off of a `release/2.1` branch, which is always the latest release candidate

Reply:
Right you are. That makes better sense to me.